### PR TITLE
self referencing connections

### PR DESCRIPTION
**Pull request by alixander**

### CODE REVIEW

File: d2graph/d2graph.go [hunk 1]
```diff
@@ -719,10 +719,6 @@ func (obj *Object) Connect(srcID, dstID []string, srcArrow, dstArrow bool, label
 	src := srcObj.EnsureChild(srcID)
 	dst := dstObj.EnsureChild(dstID)
 
-	if src == dst {
-		return nil, errors.New("self-referencing connection")
```
Comment (nhooyr) [inline d2graph/d2graph.go:723]: wait what it just worked lol by getting rid of this error?

Comment (nhooyr) [reply d2graph/d2graph.go:723]: is it only TALA that doesn't support self referencing connections?

Comment (pleshevskiy) [reply d2graph/d2graph.go:723]: Seems like dagre too... without fail but... don't render connection

Comment (alixander) [reply d2graph/d2graph.go:723]: yeah it's wip, hopefully i'll find a way to configure dagre and elk to handle this. TALA doesn't support either but we can get that going in a day or so

Comment (alixander) [reply d2graph/d2graph.go:731]: alright fixed the bug in dagre and configured elk to have more spacing. it's okayy i guess. i'll turn this on for the sake of sequence diagrams for next release

**Comment by pleshevskiy** [comment]
@alixander I had an arrow in ELK in the first version (WIP), but now it doesn't display.

# Before (WIP)

image

# After 

image

**Pull request merged**